diff --git a/tests/test-archive.t b/tests/test-archive.t
index 8670c018..1fe349fd 100644
--- a/tests/test-archive.t
+++ b/tests/test-archive.t
@@ -573,17 +573,17 @@ old file -- date clamped to 1980
   \s*Length.* (re)
   *172*80*00:00*old/.hg_archival.txt (glob)
   *0*80*00:00*old/old (glob)
 
 test xz support only available in Python 3.4
 
 #if lzma
   $ hg archive ../archive.txz
-  $ which xz >/dev/null && xz -l ../archive.txz | head -n1 || true
+  $ which xz >/dev/null 2>&1 && xz -l ../archive.txz | head -n1 || true
   Strms  Blocks   Compressed Uncompressed  Ratio  Check   Filename (xz !)
   $ rm -f ../archive.txz
 #endif
 #if py3 no-lzma
   $ hg archive ../archive.txz
   abort: lzma module is not available
   [255]
 #endif
diff --git a/tests/test-run-tests.t b/tests/test-run-tests.t
index 88adb399..df9b6c06 100644
--- a/tests/test-run-tests.t
+++ b/tests/test-run-tests.t
@@ -16,17 +16,17 @@ Define a helper to avoid the install step
   $ rt()
   > {
   >     "$PYTHON" $TESTDIR/run-tests.py --with-hg=$HGTEST_REAL_HG -j1 "$@"
   > }
 
 error paths
 
 #if symlink
-  $ ln -s `which true` hg
+  $ ln -s `which true 2>/dev/null` hg
   $ "$PYTHON" $TESTDIR/run-tests.py --with-hg=./hg
   warning: --with-hg should specify an hg script, not: true
   running 0 tests using 0 parallel processes 
   
   # Ran 0 tests, 0 skipped, 0 failed.
   $ rm hg
 #endif
 
diff --git a/tests/test-convert-cvsnt-mergepoints.t b/tests/test-convert-cvsnt-mergepoints.t
index 54416fe4..bccf25d5 100644
--- a/tests/test-convert-cvsnt-mergepoints.t
+++ b/tests/test-convert-cvsnt-mergepoints.t
@@ -97,17 +97,17 @@ return to trunk and merge MYBRANCH1_2
 
   $ cvscall -Q update -P -A
   $ filterpath cvscall -Q update -P -jMYBRANCH1_2
   RCS file: *REPO*/foo/foo.txt,v
   retrieving revision 1.1
   retrieving revision 1.1.2.2.2.1
   Merging differences between 1.1 and 1.1.2.2.2.1 into foo.txt
   $ cvsci -m "merge2" foo.txt
-  $ REALCVS=`which cvs`
+  $ REALCVS=`which cvs 2>/dev/null`
   $ echo "for x in \$*; do if [ \"\$x\" = \"rlog\" ]; then echo \"RCS file: $CVSROOT/foo/foo.txt,v\"; cat \"$TESTDIR/test-convert-cvsnt-mergepoints.rlog\"; exit 0; fi; done; $REALCVS \$*" > ../cvs
   $ chmod +x ../cvs
   $ PATH=..:${PATH} hg debugcvsps --parents foo
   collecting CVS rlog
   7 log entries
   creating changesets
   7 changeset entries
   ---------------------
