---
 hgdemandimport/__init__.py |    3 +++
 1 file changed, 3 insertions(+)

--- a/hgdemandimport/__init__.py
+++ b/hgdemandimport/__init__.py
@@ -53,16 +53,19 @@ IGNORES = {
     'distutils.msvc9compiler',
     '__builtin__',
     'builtins',
     'urwid.command_map',  # for pudb
     'lzma',
     # setuptools uses this hack to inject it's own distutils at import time
     'setuptools',
     '_distutils_hack.override',
+    # local import in importlib
+    '_weakrefset',
+    'threading',
 }
 
 _pypy = '__pypy__' in sys.builtin_module_names
 
 if _pypy:
     # _ctypes.pointer is shadowed by "from ... import pointer" (PyPy 5)
     IGNORES.add('_ctypes.pointer')
     # pure Python module on PyPy, must be loaded to raise ModuleNotFoundError
